runtime/type.js

```diff
@@ -33,12 +33,14 @@ function addType(name, tag, args) {
         return new Type(tag, data);
       }});
     for (var arg of args) {
-      var upperArg = arg[0].toUpperCase() + arg.substring(1);
-      Object.defineProperty(Type.prototype, `${lowerName}${upperArg}`, {
-        get: function() {
-          assert(this[`is${name}`]);
-          return this.data[arg];
-        }});
+      (function(arg) {
```

Comment:
I think `for (let arg of args)` does what you want here.

Reply:
done